Framework/Algorithms/src/ResizeRectangularDetector.cpp

```diff
@@ -97,19 +97,19 @@ void ResizeRectangularDetector::exec() {
     throw std::runtime_error("Component with name " + ComponentName +
                              " was not found.");
 
-  auto &componentInfo = inputW->mutableComponentInfo();
-  const auto componentIndex = componentInfo.indexOf(comp->getComponentID());
-  if (!componentInfo.isDetector(componentIndex))
+  RectangularDetector_const_sptr det =
+      boost::dynamic_pointer_cast<const RectangularDetector>(comp);
```

Comment:
I wouldn't expect that to be done here though. It might be an additional refactoring step.